heudiconv/dicoms.py

```diff
@@ -0,0 +1,507 @@
+            key = studyUID.split('.')[-1]
+        elif per_accession_number:
+            key = accession_number
+        else:
+            key = ''
+        lgr.debug("%30s %30s %27s %27s %5s nref=%-2d nsrc=%-2d %s" % (
+            key,
+            info.series_id,
+            dcminfo.SeriesDescription,
+            dcminfo.ProtocolName,
+            info.is_derived,
+            len(dcminfo.get('ReferencedImageSequence', '')),
+            len(dcminfo.get('SourceImageSequence', '')),
+            info.image_type
+        ))
+        if per_studyUID:
+            if studyUID not in seqinfo:
+                seqinfo[studyUID] = OrderedDict()
+            seqinfo[studyUID][info] = series_files
+        elif per_accession_number:
+            if accession_number not in seqinfo:
+                seqinfo[accession_number] = OrderedDict()
+            seqinfo[accession_number][info] = series_files
+        else:
+            seqinfo[info] = series_files
+
+    if per_studyUID:
+        lgr.info("Generated sequence info for %d studies with %d entries total",
+                 len(seqinfo), sum(map(len, seqinfo.values())))
+    elif per_accession_number:
+        lgr.info("Generated sequence info for %d accession numbers with %d "
+                 "entries total", len(seqinfo), sum(map(len, seqinfo.values())))
+    else:
+        lgr.info("Generated sequence info with %d entries", len(seqinfo))
+    return seqinfo
+
+
+def get_dicom_series_time(dicom_list):
+    """Get time in seconds since epoch from dicom series date and time
+    Primarily to be used for reproducible time stamping
+    """
+    import time
+    import calendar
+
+    dcm = dcm.read_file(dicom_list[0], stop_before_pixels=True, force=True)
+    dcm_date = dcm.SeriesDate  # YYYYMMDD
+    dcm_time = dcm.SeriesTime  # HHMMSS.MICROSEC
+    dicom_time_str = dcm_date + dcm_time.split('.', 1)[0]  # YYYYMMDDHHMMSS
+    # convert to epoch
+    return calendar.timegm(time.strptime(dicom_time_str, '%Y%m%d%H%M%S'))
+
+
+def compress_dicoms(dicom_list, out_prefix, tempdirs):
+    """Archives DICOMs into a tarball
+
+    Also tries to do it reproducibly, so takes the date for files
+    and target tarball based on the series time (within the first file)
+
+    Parameters
+    ----------
+    dicom_list : list of str
+      list of dicom files
+    out_prefix : str
+      output path prefix, including the portion of the output file name
+      before .dicom.tgz suffix
+    tempdirs : object
+      TempDirs object to handle multiple tmpdirs
+
+    Returns
+    -------
+    filename : str
+      Result tarball
+    """
+    # if tempdirs:
+    tmpdir = tempdirs(prefix='dicomtar')
+    # else:
+    #     tmpdir = mkdtemp(prefix='dicomtar')
+    outtar = out_prefix + '.dicom.tgz'
+    if op.exists(outtar): # and not global_options['overwrite']:
+        raise RuntimeError("File %s already exists, will not override, ensure "
+                           "unique scan names" % outtar)
```

Comment:
so "overwrite" mode is disabled now?

Reply:
nope - just forgot to add the argparse option to this